scripts/superdesk-authoring/metadata/metadata.js

```diff
@@ -470,23 +499,29 @@ function MetadataListEditingDirective(metadata, $filter, $timeout) {
                         //build object
                         var o = {};
                         o[scope.field] = t;
-                        _.extend(scope.item, o);
                     }
 
-                    scope.selectedTerm = '';
-
-                    if (!reloadList) {
-                        // Remove the selected term from the terms
-                        scope.terms = _.without(scope.terms, term);
-                        scope.activeTree = _.without(scope.activeTree, term);
-                    }
+                    // wait for pending request before applying change (e.g: while autosaving)
+                    var waitForPendingRequests = function() {
```

Comment:
what is the idea here? wait for autosave to finish? then it would be better imo to broadcast an event from autosave..
also seems like there is some code duplication later